File: C++/Easy/Accepted/sort-integers-by-the-number-of-1-bits_4.cpp
class Solution {
public:
    vector<int> sortByBits(vector<int>& arr) {
        
        sort(arr.begin(), arr.end(), [](int a, int b) {
            int cntA = 0, cntB = 0;
            for (int i = 0; i < 14; i++) {
                if (a & (1 << i)) cntA++;
                if (b & (1 << i)) cntB++;
            }
            if (cntA == cntB) {
                return a < b;
            } else {
                return cntA < cntB;
            }
        });
        return arr;
    }
};

File: Java/Easy/CompileError/sort-integers-by-the-number-of-1-bits_4.java
class Solution {
    public int[] sortByBits(int[] arr) {
        
        int[] bits = new int[10001];
        for (int i = 0; i < 10001; i++) {
            bits[i] = countBits(i);
        }
        Arrays.sort(arr);
        Arrays.sort(arr, (a, b) -> bits[a] - bits[b]);
        return arr;
    }

    private int countBits(int x) {
        int count = 0;
        while (x > 0) {
            if ((x & 1) == 1) {
                count++;
            }
            x >>= 1;
        }
        return count;
    }
}